gateway/utils/jwt/jwt.go

```diff
@@ -53,11 +55,13 @@ func (j *JWT) SetSecrets(secrets []*config.Secret) error {
 	newKidMap := map[string]string{}
 	newStaticSecretMap := map[string]*config.Secret{}
 	for _, secret := range secrets {
-		if secret.KID == "" {
-			secret.KID = ksuid.New().String()
-		}
 		switch secret.Alg {
 		case config.JwkURL:
+			// Set the secret kid if it isn't already set
+			if secret.KID == "" {
+				secret.KID = ksuid.New().String()
+			}
+
```

Comment:
Is this required? 
as we create sha 265 id below

Reply:
This is for jwk. The rest is for different alg.